spark3/src/main/java/org/apache/iceberg/spark/BaseCatalog.java

```diff
@@ -45,4 +45,6 @@ public Procedure loadProcedure(Identifier ident) throws NoSuchProcedureException
 
     throw new NoSuchProcedureException(ident);
   }
+
+  protected abstract boolean dropTableWithPurge(Identifier ident, boolean purge);
```

Comment:
This doesn't need to be an abstract method on this class. Both implementations of `BaseCatalog` implement `dropTableWithPurge` and it should just be private in both of those.